hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieTableServiceClient.java

```diff
@@ -821,16 +821,8 @@ protected void archive(HoodieTable table) {
    * @return
    */
   private HoodieTimeline getInflightTimelineExcludeCompactionAndClustering(HoodieTableMetaClient metaClient) {
-    HoodieTimeline inflightTimelineWithReplaceCommit = metaClient.getCommitsTimeline().filterPendingExcludingCompaction();
-    HoodieTimeline inflightTimelineExcludeClusteringCommit = inflightTimelineWithReplaceCommit.filter(instant -> {
-      if (instant.getAction().equals(HoodieTimeline.REPLACE_COMMIT_ACTION)) {
-        Option<Pair<HoodieInstant, HoodieClusteringPlan>> instantPlan = ClusteringUtils.getClusteringPlan(metaClient, instant);
-        return !instantPlan.isPresent();
-      } else {
-        return true;
-      }
-    });
-    return inflightTimelineExcludeClusteringCommit;
+    HoodieTimeline inflightTimelineExcludingCompaction = metaClient.getCommitsTimeline().filterPendingExcludingCompaction();
+    return inflightTimelineExcludingCompaction.filter(instant -> !ClusteringUtils.isClusteringInstant(inflightTimelineExcludingCompaction, instant));
```

Comment:
won't this be simplified w/ this patch. 
inflightTimelineExcludingCompaction will only include pending instants and not compaction. 
so, we might as well just filter out CLUSTER_ACTION right? we do not need to deser the requested to find if its clustering. 

infact thats one of the major wins w/ this change.

Reply:
> For backwards compatibility purpose, we had to consider pending replace commits also in addition to pending clustering commits. So, lets ensure to check both timeline for now.

If we are going to check both timeline, then this would remain as it is? Similarly for the other comment.